userspace/libsinsp/event.cpp

```diff
@@ -1402,6 +1402,34 @@ Json::Value sinsp_evt::get_param_as_json(uint32_t id, OUT const char** resolved_
 	return ret;
 }
 
+string sinsp_evt::get_cwd(uint32_t id, sinsp_threadinfo* tinfo)
+{
+	auto cwd = tinfo->get_cwd();
+
+	auto param = &(m_params[id]);
+	auto payload = param->m_val;
+
+	// If pathname is relative (does not start with a "/")
+	if (strncmp(payload, "/", 1) != 0) {
+		// Get the previous parameter
+		auto prev_id = id - 1;
```

Comment:
You should verify that `id` is nonzero.

I'm not overly familiar with the sinsp event param parsing code, but I think it's possible for there to be a `FSPATH` parameter at index 0. If that is the case, `id - 1` will underflow and then read well past the end of the array.